### PR TITLE
FIREFLY-976: fix bug introduced in useStoreConnector and fix file upload dialog

### PR DESCRIPTION
**Pull request by loitly**
Regression bugs reported after FIREFLY-976 was merged.  Please test and confirmed it's fixed.

https://firefly-976-tablepanel-rerender.irsakudev.ipac.caltech.edu/applications/sofia/
https://fireflydev.ipac.caltech.edu/firefly-976-tablepanel-rerender/firefly/

### CODE REVIEW

File: src/firefly/js/ui/SimpleComponent.jsx [hunk 1]
```diff
@@ -60,12 +60,14 @@ export function useStoreConnector(stateGetter, deps=[]) {
 
     let isMounted = true;
     useEffect(() => {
+        let cState = val;
```
Comment (robyww) [inline src/firefly/js/ui/SimpleComponent.jsx:63]: Well I think i understand why this line fixes it but it is tricky. This is a double closure with `val` is not really the old value.

File: src/firefly/js/ui/SimpleComponent.jsx [hunk 2]
```diff
@@ -60,12 +60,14 @@ export function useStoreConnector(stateGetter, deps=[]) {
...
             if (isMounted) {
-                const newState = stateGetter(val);      // if getter returns oldState then no state update
-                if (newState===val) return;             // comparator might be overridden, use === first for efficiency
-                if ( !comparator(val, newState) ) {
-                    setter(newState);
+                const nState = stateGetter(val);      // if getter returns oldState then no state update
+                if (nState===cState) return;             // comparator might be overridden, use === first for efficiency
+                if ( !comparator(cState, nState) ) {
+                    cState = nState;
+                    setter(cState);
```
Comment (wmiipac) [inline src/firefly/js/ui/SimpleComponent.jsx:70]: I had to get explanation from Loi,  this change make sense.

**Review by wmiipac — APPROVED** [review]
Tested all affected apps, all look good!

**Review by robyww — APPROVED** [review]
I tested the 3 color tab issue I was aware of it it works.

Looks fixed.

File: src/firefly/js/ui/SimpleComponent.jsx (hunk 1 above)
Comment (wmiipac) [reply src/firefly/js/ui/SimpleComponent.jsx:63]: I didn't get it, I went to Loi's office with ??? "double closure", it is!

**Pull request merged**